frontend/src/AddTagView.svelte

```diff
@@ -0,0 +1,222 @@
+    align-content: center;
+    white-space: nowrap;
+  }
+
+  .nowrap {
+    white-space: nowrap;
+  }
+
+  .error {
+    margin-top: 2em;
+  }
+
+  h1 {
+    text-transform: uppercase;
+    font-weight: bold;
+    color: inherit;
+    font-size: xxx-large;
+    line-height: 1;
+    margin: 0;
+    max-width: 50%;
+    text-align: center;
+  }
+
+  form {
+    max-width: 50%;
+  }
+
+  .center {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    width: 100%;
+    height: 100%;
+    color: var(--main-fg-color);
+  }
+
+  button,
+  select,
+  option {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    padding-top: 0.5em;
+    padding-bottom: 0.5em;
+    padding-left: 1em;
+    padding-right: 1em;
+    margin-left: 0.5em;
+    margin-right: 0.5em;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  option {
+    font-family: monospace;
+  }
+
+  input[type="file"] {
+    display: none;
+  }
+</style>
+
+<script>
+  import { selected, selectedResource } from "./stores.js";
+  import { onMount } from "svelte";
+  import LoadingText from "./LoadingText.svelte";
+  import { cleanOfrakType } from "./helpers";
+
+  export let modifierView, resourceNodeDataMap, dataPromise;
+  let errorMessage,
+    ofrakTagsPromise = new Promise(() => {}),
+    selectedTag;
+
+  function refreshResource() {
+    // Force tree view children refresh
+    resourceNodeDataMap[$selected].collapsed = false;
+    resourceNodeDataMap[$selected].childrenPromise =
+      $selectedResource.get_children();
+
+    // Force hex view refresh with colors
+    const originalSelected = $selected;
+    $selected = undefined;
+    $selected = originalSelected;
+  }
+
+  function chooseTag() {
+    if (selectedTag) {
+      modifierView = undefined;
+      $selectedResource.add_tag(selectedTag);
+    }
+  }
+
+  onMount(async () => {
+    try {
+      await fetch(`/get_all_tags`).then(async (r) => {
+        if (!r.ok) {
+          throw Error(JSON.stringify(await r.json(), undefined, 2));
+        }
+        r.json().then((ofrakTags) => {
+          ofrakTags.sort(function (a, b) {
+            if (cleanOfrakType(a) > cleanOfrakType(b)) {
+              return 1;
+            } else {
+              return -1;
+            }
+          });
+          ofrakTagsPromise = ofrakTags;
+        });
+      });
+    } catch (err) {
+      try {
+        errorMessage = JSON.parse(err.message).message;
+      } catch (_) {
+        errorMessage = err.message;
+      }
+    }
+  });
+</script>
+
+<div class="container">
+  <div class="inputs">
+    <p>Select tag to add to resource.</p>
+    New tag:
+    {#await ofrakTagsPromise}
+      <LoadingText />
+    {:then ofrakTags}
+      {#if ofrakTags && ofrakTags.length > 0}
+        <form on:submit|preventDefault="{chooseTag}">
+          <select
```

Comment:
I think this would look cleaner if the `<form>` was a row flexbox with the `<select>` having `flex-grow: 1`. That way the label, dropdown, and button would all be in a row, and would take up the full width of the containing element. When doing this, removing the `max-width: 50%` from the `<form>` will help as well.

```suggestion
    {#await ofrakTagsPromise}
      <LoadingText />
    {:then ofrakTags}
      {#if ofrakTags && ofrakTags.length > 0}
        <form on:submit|preventDefault="{chooseTag}">
          New tag:
          <select
```

If you'd rather, I can just make those changes and push them to this branch.

Reply:
I tweaked it slightly so the "New Tag" text is inline with that and there's no more max width. But I don't understand the flex grow (can't even add it anyway as the parent's `display: block` style is apparently incompatible) and it looks fine.

Reply:
I just pushed a change that addresses the remainder of this comment.